core/unit_test/TestLegionInitialization.cpp

```diff
@@ -0,0 +1,89 @@
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <Kokkos_Core.hpp>
+#include <gtest/gtest.h>
+
+namespace {
+
+struct ReductionFunctor {
+  Kokkos::View<int*> d;
+
+  KOKKOS_FUNCTION void operator()(int i, int& sum) const { sum += d(i); }
+};
+
+// The purpose of this test is to mimic Legion's use case of initializing and
+// finalizing individual backends
+TEST(initialization, legion_initialization) {
+  Kokkos::InitializationSettings kokkos_init_settings;
+
+  Kokkos::Impl::pre_initialize(kokkos_init_settings);
+
+  // We need to have a host execution space initialized first.
+  Kokkos::HostSpace::execution_space::impl_initialize(kokkos_init_settings);
+
+  if (!std::is_same_v<Kokkos::DefaultExecutionSpace,
+                      Kokkos::HostSpace::execution_space>)
+    Kokkos::DefaultExecutionSpace::impl_initialize(kokkos_init_settings);
+
+  Kokkos::Impl::post_initialize(kokkos_init_settings);
+
+  {
+    Kokkos::View<int*> d("d", 1000);
+    Kokkos::deep_copy(d, 1);
+    int result;
+    Kokkos::parallel_reduce("TestRed", d.extent(0), ReductionFunctor{d},
+                            result);
+    EXPECT_EQ(result, d.extent(0));
+  }
+
+  Kokkos::DefaultExecutionSpace::impl_finalize();
+  Kokkos::finalize();
+}
+}  // namespace
+
+int main(int argc, char* argv[]) {
+  ::testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
```

Comment:
We already have a version of main that does that.
`core/unit_test/UnitTestMain.cpp`
Any particular reason why we should not use it there too?